genie-agent/src/main/java/com/netflix/genie/agent/execution/services/impl/grpc/GRpcAgentFileStreamServiceImpl.java

```diff
@@ -379,30 +410,34 @@ private void completeTransfer(final boolean shutdownStream, @Nullable final Exce
         private void sendChunk() throws IOException {
 
             if (this.watermark < this.endOffset - 1) {
```

Comment:
Does this class need to be thread safe? If so should these variables be atomic?

Reply:
The parent class is synchronized where appropriate.
This inner `FileTransfer` class should only be used by one (gRPC) thread at the time so adding further synchronization should not be necessary.